plugin/hydrate_error.go

```diff
@@ -47,6 +54,57 @@ func RetryHydrate(ctx context.Context, d *QueryData, hydrateData *HydrateData, h
 	return hydrateResult, err
 }
 
+func getBackoff(retryConfig *RetryConfig) (retry.Backoff, error) {
```

Comment:
could you do something like 
```
func getBackoff(retryConfig *RetryConfig) (retry.Backoff, error) {
	// Default set to Fibonacci
	backoffAlgorithm := "Fibonacci"
	retryInterval := time.Duration(100) // in ms

	// Check from config
	if retryConfig != nil {
		if retryConfig.BackoffAlgorithm != "" {
			backoffAlgorithm = retryConfig.BackoffAlgorithm
		}

		if retryConfig.RetryInterval != 0 {
			retryInterval = time.Duration(retryConfig.RetryInterval)
		}
		if retryConfig.CappedDuration != 0 {
			backoffAlgorithm = "CappedDuration"
		}

		if retryConfig.MaxDuration != 0 {
			backoffAlgorithm = "CappedDuration"
		}
	}

	var backoff retry.Backoff
	var err error
	switch backoffAlgorithm {
	case "Fibonacci":
		backoff, err = retry.NewFibonacci(retryInterval * time.Millisecond)
		if err != nil {
			return nil, err
		}
	case "Exponential":
		backoff, err = retry.NewExponential(retryInterval * time.Millisecond)
		if err != nil {
			return nil, err
		}
	case "Constant":
		backoff, err = retry.NewConstant(retryInterval * time.Millisecond)
		if err != nil {
			return nil, err
		}
	case "CappedDuration":
		backoff = retry.WithCappedDuration(time.Duration(retryConfig.CappedDuration)*time.Millisecond, backoff)
	case "MaxDuration":
		backoff = retry.WithMaxDuration(time.Duration(retryConfig.MaxDuration)*time.Second, backoff)
	}

	return backoff, nil
}
```

Reply:
@kaidaguerre `CappedDuration` and `MaxDuration` are used to apply caps or limits on the backoff algorithms.
If we use the following code
```
if retryConfig.CappedDuration != 0 {
	backoffAlgorithm = "CappedDuration"
}

if retryConfig.MaxDuration != 0 {
	backoffAlgorithm = "CappedDuration"
}
```
It will directly go to either `case "CappedDuration":` or `case "MaxDuration":`, which is not correct.